scapy/layers/usb.py

```diff
@@ -284,3 +287,206 @@ def close(self):
+
+    .. seealso::
+    Source - https://github.com/the-tcpdump-group/libpcap/blob/ba0ef0353ed9f9f49a1edcfb49fefaf12dec54de/pcap/usb.h#L94
+
+
+    .. seealso::
+    Source - https://www.kernel.org/doc/Documentation/usb/usbmon.txt
+
+
+    .. seealso::
+    Source - https://www.kernel.org/doc/html/latest/driver-api/usb/URB.html
+
+
+    .. seealso::
+    Source - https://wiki.wireshark.org/USB
+    """
+
+    HEADER_SIZE = None
+    name = "USBMonHeader"
+
+    class EventType(IntEnum):
+        completion = ord('C')
+        error = ord('E')
+        submit = ord('S')
+
+    class TransferType(IntEnum):
+        isochronous = 0
+        interrupt = 1
+        control = 2
+        bulk = 3
+
+    class SetupFlag(IntEnum):
+        relevant = 0
+        irrelevant = ord("-")
+
+    class DataFlag(IntEnum):
+        urb = 0
+        incoming = ord("<")
+        outgoing = ord(">")
+        error = ord("E")
+
+    class TimeStamp(Packet):
+        name = "TimeStamp"
+
+        @staticmethod
+        def _getSize(*_args):
+            return 12
+
+        fields_desc = [
+            LELongField("seconds", None),  # 8
+            LEIntField("microseconds", None),  # 4
+        ]
+
+    @property
+    def needs_setup(self):
+        SetupFlag = self.__class__.SetupFlag
+        return SetupFlag(self.setup_flag) == SetupFlag.relevant
+
+    @property
+    def is_isochr(self):
+        TransferType = self.__class__.TransferType
+        return TransferType(self.transfer_type) == TransferType.isochronous
+
+    HEADER_STATIC_PART_SIZE = 8 + 1 + 1 + 1 + 1 + 2 + 1 + 1 + TimeStamp._getSize() + 4 + 4 + 4  # pylint:disable=protected-access
+
+    def _getOptionalPartSize(pkt=None):
+        return 24
+
+    def _getPaddingSize(pkt):
+        res = pkt.__class__.HEADER_SIZE - __class__._getOptionalPartSize() * int(pkt.needs_setup) - __class__.HEADER_STATIC_PART_SIZE  # pylint:disable=protected-access
+        return res
+
+    fields_desc = [
+        LELongField("urb_id", None),  # 8
+        enumToScapyEnum("event_type", 0, EventType, "<B"),  # 1
+        enumToScapyEnum("transfer_type", 0, TransferType, "<B"),  # 1
+        EndpointNumber("endpoint_number", 0),  # 1
+        XByteField("device_address", None),  # 1
+        LEShortField("bus_id", None),  # 2
+        enumToScapyEnum("setup_flag", 0, SetupFlag, "<B"),  # 1
+        enumToScapyEnum("data_flag", 0, DataFlag, "<B"),  # 1
+        # just PacketField doesn't work and breaks everything after it
+        PacketLenField("timestamp", None, TimeStamp, length_from=TimeStamp._getSize),  # 12, pylint:disable=protected-access
+        LEIntEnumField("status", 0x0, _usbd_status_codes),  # 4
+        LEIntField("urb_size", 0),  # 4
+        LenField("data_size", 0, fmt="<i"),  # 4
+        ConditionalField(
+            MultipleTypeField(
+                [
+                    # just PacketField doesn't work and breaks everything after it
```

Comment:
What do you mean?
You're likely missing a `guess_default_payload` in your `SetupIsocr`. See for instance https://github.com/secdev/scapy/blob/e457fd161ea8111d7d7c08625e333d0f12ee51cc/scapy/layers/ntlm.py#L374-L375